index.js

```diff
@@ -83,6 +99,133 @@ function attachToBody (options, req, reply, next) {
   })
 }
 
+const allowedTokens = ['[', ']', '{}']
+
+function getSerializedKeyTokens (key) {
+  const tokens = key.split(/(\[|]|{})/).filter(Boolean)
+  return tokens.some(token => allowedTokens.includes(token)) ? tokens : []
+}
+
+function validateSerializedKey (tokens, value) {
+  let isValid = true
+  let token = tokens[0]
+
+  // first token can't be '[', ']', or '{}'
+  if (allowedTokens.includes(token)) isValid = false
+  else if (token === '__proto__' || token === 'constructor') {
+    return `${token} is not allowed as a key name`
+  } else {
+    const bracesEndingCount = tokens.filter(t => t === '{}').length
```

Comment:
this is definetly slower than a for loop

Reply:
I haven't benchmarked it on Node, but [this is what I got on Chrome 112.0.5615.137](https://perf.link/#eyJpZCI6InN1NzRlZ3Q1NjJsIiwidGl0bGUiOiJHZXR0aW5nIGl0ZW0gY291bnQ6IGZpbHRlciBWUyBmb3IgVlMgd2hpbGUiLCJiZWZvcmUiOiJjb25zdCBkYXRhID0gWydwcm9wQVtwcm9wQUFdJywgJ3Byb3BBW3Byb3BBQl0nLCAncHJvcEFbcHJvcEFDXVtdJywgJ3Byb3BBW3Byb3BBQ11bXScsICdwcm9wQVtwcm9wQUNdW10nLCAncHJvcEJbMF1bcHJvcEJBXVtwcm9wQkFBXScsICdwcm9wQlswXVtwcm9wQkFdW3Byb3BCQUJdJywgJ3Byb3BCWzBdW3Byb3BCQV1bcHJvcEJBQ11bMF0nLCAncHJvcEJbMF1bcHJvcEJBXVtwcm9wQkFDXVsxXScsICdmb297fVthYWFde31bYmFyXScsICdwcm9wQlswXVtwcm9wQkFdW3Byb3BCQUNdWzJdJywgJ3Byb3BCWzBdW3Byb3BCQl0nLCAncHJvcEJbMV1bcHJvcEJBXVtwcm9wQkFBXScsICdwcm9wQ1tdJywgJ3Byb3BDW10nLCAncHJvcENbXScsICdwcm9wRCcsICdwcm9wRScsICdwcm9wRScsICdwcm9wRScsICdwcm9wRnt9JywgJ3Byb3BHWzBdW3Byb3BHQV1bXXt9JywgJ3Byb3BHWzFde30nLCAnYWFhe317fSddOyIsInRlc3RzIjpbeyJuYW1lIjoiZm9yIiwiY29kZSI6ImxldCBicmFjZXNFbmRpbmdDb3VudCA9IDBcbmxldCB0b2tlbjtcblxuZm9yIChsZXQgaSA9IDA7IGkgPCBkYXRhLmxlbmd0aDsgaSsrKSB7XG4gIHRva2VuID0gZGF0YVtpXVxuICBpZiAodG9rZW4gPT09ICd7fScpIHtcbiAgICBicmFjZXNFbmRpbmdDb3VudCsrXG4gIH1cbn0iLCJydW5zIjpbMjMzMDAwLDI4MjAwMCwxOTQwMDAsMzAwMDAwLDMyOTAwMCwyMDgwMDAsMzQ4MDAwLDE0MzAwMCwyODMwMDAsMCwxNjkwMDAsMzYzMDAwLDgwMDAsMzAwMDAwLDcxNjAwMCwyODAwMCwxNzEwMDAsNDYwMDAsNTUyMDAwLDI4NjAwMCwyNzUwMDAsODMxMDAwLDIwNzAwMCw1MTIwMDAsNTEzMDAwLDMxNzAwMCwxODEwMDAsMjMwMDAwLDYxMDAwLDExNzAwMCwzMzcwMDAsNTc5MDAwLDE5NjAwMCwzMTQwMDAsMzQ5MDAwLDI4MjAwMCw2NDUwMDAsMTkxMDAwLDE4MTAwMCwyNjIwMDAsMzAwMDAsMzcwMDAsMzAwMDAwLDUwOTAwMCw3MjAwMDAsMzAwMDAwLDEwMDAsNDQ0MDAwLDQwNzAwMCw2MTAwMCwzMDAwMDAsNjM3MDAwLDUwMDAwLDYwMDAwMCwzMDAwLDI2OTAwMCw2NDEwMDAsMzUwMDAwLDQwMTAwMCwzMTAwMCw1NjAwMCw1MjAwMCwyOTEwMDAsNDI5MDAwLDEwOTAwMCwzMDAwMDAsMjY0MDAwLDQxODAwMCwxMTEwMDAsMzAwMDAwLDU5MDAwLDM4OTAwMCwyNzcwMDAsNzIwMDAsMzAwMDAwLDc1MzAwMCwzMDAwMDAsNDg1MDAwLDI4NTAwMCwyNzMwMDAsMjQxMDAwLDE5MDAwLDU3MjAwMCwzMDAwMDAsMzAwMDAwLDc5MDAwLDQxMjAwMCw3NjAwMCwzMDAwMDAsMTY4MDAwLDgzMDAwLDgwMDAsNDI4MDAwLDI0MTAwMCw2ODAwMCwyNDAwMDAsMjMzMDAwLDEzMzAwMCwyMDMwMDAsNTQ3MDAwXSwib3BzIjoyNzgwNDB9LHsibmFtZSI6ImZpbHRlciIsImNvZGUiOiJkYXRhLmZpbHRlcih0ID0%2BIHQgPT09ICd7fScpLmxlbmd0aCIsInJ1bnMiOls3MzAwMDAsOTAwMDAsMjkzMDAwLDI3NDAwMCw1MjAwMCw3MTMwMDAsNTE4MDAwLDI1OTAwMCwzODgwMDAsMjMwMDAwLDIzNTAwMCw1NzQwMDAsMjg4MDAwLDY0MDAwMCw4NTYwMDAsMzc4MDAwLDE5NjAwMCw1MjAwMCw3NTkwMDAsMzExMDAwLDMwMDAwMCwzNzkwMDAsODUwMDAwLDY1MzAwMCwxODAwMCw4NDAwMCw5NTAwMCw2ODEwMDAsNjE2MDAwLDcwNDAwMCw3MDAwLDUyMTAwMCwxMjkwMDAsMjY0MDAwLDM2MDAwMCwyOTUwMDAsNzAxMDAwLDU4MDAwLDI4ODAwMCwzODAwMDAsMjk5MDAwLDEzNTAwMCwyMDMwMDAsNjkxMDAwLDczMTAwMCw0MTIwMDAsNDQwMDAsNjA0MDAwLDM0MjAwMCwxOTAwMCw4NDQwMDAsMjY2MDAwLDc0ODAwMCwxODAwMCw2NTcwMDAsNDYxMDAwLDY5MzAwMCwzOTYwMDAsMTcxMDAwLDYxMjAwMCw2MTAwMCwyMTAwMCwzMzIwMDAsNzY3MDAwLDIyNzAwMCw0MDYwMDAsNDc4MDAwLDU4MzAwMCw3MDAwMCwzOTQwMDAsNDYwMDAsODk1MDAwLDE0OTAwMCw2NzAwMCwyNTMwMDAsMjUyMDAwLDQ3ODAwMCwxMjAwMDAsNTIwMDAsNjE0MDAwLDg1MjAwMCwyMTAwMCw3MDAwMDAsMzkxMDAwLDc4NTAwMCwxMTAwMCw3NzMwMDAsODczMDAwLDQ3MDAwMCw3ODQwMDAsMzAwMCwxMzAwMCw0NzcwMDAsMTkwMDAsNTQ4MDAwLDUzODAwMCw2NDYwMDAsMTY1MDAwLDMwNDAwMCwxMDQwMDBdLCJvcHMiOjM4MzA3MH0seyJuYW1lIjoibmVnYXRpdmUgd2hpbGUiLCJjb2RlIjoibGV0IGkgPSBkYXRhLmxlbmd0aCAtIDFcbmxldCB0b2tlbiA9IGRhdGFbaV1cbmxldCBicmFjZXNFbmRpbmdDb3VudCA9IDBcblxud2hpbGUgKGkgPj0gMCkge1xuICBpZiAodG9rZW4gPT09ICd7fScpIHtcbiAgICBicmFjZXNFbmRpbmdDb3VudCsrXG4gIH1cbiAgdG9rZW4gPSBkYXRhWy0taV1cbn0iLCJydW5zIjpbOTYwMDAsNjEwMDAwLDI3MDAwMCw2MDEwMDAsNDAwMDAsNDI0MDAwLDM0NDAwMCw4NDAwMCwyNDUwMDAsNDU3MDAwLDE1ODAwMCwyODQwMDAsMzk2MDAwLDMyNTAwMCw1NjUwMDAsMzk2MDAwLDE0NjAwMCwzNTIwMDAsMzA3MDAwLDU0MDAwMCw0ODkwMDAsMzY1MDAwLDM3MDAwLDI3MDAwMCwyNjIwMDAsMzk2MDAwLDY2MDAwLDM5NjAwMCwzMjIwMDAsODAwMCwxNTUwMDAsMTUwMDAsMTQ3MDAwLDM3NjAwMCwyMDQwMDAsNTkwMDAsMjAwMCw1NDkwMDAsMTA2MDAwLDk0MDAwLDM1MjAwMCw4NzAwMCwxNDAwMDAsNDQyMDAwLDU1NzAwMCwzOTgwMDAsMjMwMDAsMjIwMDAwLDE3MjAwMCw1NzAwMDAsMTczMDAwLDEwMTAwMCwyMjUwMDAsMzc2MDAwLDI1NjAwMCwzMTkwMDAsMjUxMDAwLDgxMDAwLDUwNTAwMCwyMzMwMDAsMzE3MDAwLDE5MzAwMCwzOTYwMDAsNDk5MDAwLDQyMDAwLDIzMTAwMCwyODAwMDAsMzk2MDAwLDYwMjAwMCwzOTkwMDAsMzc1MDAwLDM5NjAwMCw2NDIwMDAsNjY2MDAwLDM3MzAwMCwyNzUwMDAsNDEwMDAwLDMyOTAwMCwyODEwMDAsMjAwMCw1MzkwMDAsNTY1MDAwLDQxNjAwMCwzMTcwMDAsMzk2MDAwLDM5NjAwMCwzNzQwMDAsMzEyMDAwLDY0MjAwMCw0NTkwMDAsMjgwMDAsNTA1MDAwLDM5NjAwMCw1MzAwMCwzMzAwMCw3OTAwMCwyNTAwMDAsMzYwMDAsMTkwMDAwLDk1MDAwXSwib3BzIjoyOTYyNDB9XSwidXBkYXRlZCI6IjIwMjMtMDUtMTJUMTk6MDg6NDYuMTQ4WiJ9). I've replaced it with a negative `while` anyway, as I noticed that array methods like `filter`, `find`, `some` or `every` are not used in the project.